pallets/dapp-staking-v3/src/types.rs

```diff
@@ -227,11 +232,45 @@ where
 #[derive(Encode, Decode, MaxEncodedLen, Clone, Copy, Debug, PartialEq, Eq, TypeInfo)]
 pub enum PeriodType {
     /// Period during which the focus is on voting.
-    /// Inner value is the era in which the voting period ends.
-    Voting(#[codec(compact)] EraNumber),
+    Voting,
     /// Period during which dApps and stakers earn rewards.
-    /// Inner value is the era in which the Build&Eearn period ends.
-    BuildAndEarn(#[codec(compact)] EraNumber),
+    BuildAndEarn,
+}
+
+impl PeriodType {
+    pub fn next(&self) -> Self {
+        match self {
+            PeriodType::Voting => PeriodType::BuildAndEarn,
+            PeriodType::BuildAndEarn => PeriodType::Voting,
+        }
+    }
+}
+
+/// Wrapper type around current `PeriodType` and era number when it's expected to end.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, Copy, Debug, PartialEq, Eq, TypeInfo)]
+pub struct PeriodInfo {
+    #[codec(compact)]
+    pub number: PeriodNumber,
```

Comment:
Docstring to explain the role of period number would help here. For me it's not very clear right now.

Reply:
I can add that, sure. But I'm not sure how can I explain it?
It's the period number; number of the ongoing period.

Perhaps the broader context is missing when only looking at this struct.
Later I will add a more detailed explanation of the functionality into the PR.
For now, I'd suggest to refer to the forum post and/or design docs.

If you have a good suggestions, I'll add that.